lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1515,11 +1636,11 @@ def _update_relation_data(self, _):
         if not self._charm.unit.is_leader():
             return
 
-        alert_rules = AlertRules(self.topology)
-        alert_rules.add_path(self.dir_path, recursive=self._recursive)
+        alert_rules = AlertRules()
+        path_added = alert_rules.add_path(self.dir_path, recursive=self._recursive)
```

Comment:
`set_topology` missing?
The `set_topology` shenanigans remind me of [RAII](https://en.cppreference.com/w/cpp/language/raii).
AFAIU, 
- `AlertRules` cannot exist without topology.
- if we want to render alert rules using multiple topologies we can implement `__add__`.